src/Raven.Client/Http/RavenCommand.cs

```diff
@@ -84,7 +84,7 @@ public bool IsFailedWithNode(ServerNode node)
         {
             using (response)
             {
-                if (ResponseType == RavenCommandResponseType.Empty)
+                if (ResponseType == RavenCommandResponseType.Empty || response.StatusCode == HttpStatusCode.NoContent)
```

Comment:
We are using ExecuteJsonAsync, which uses JsonCommandWithPayload, and it uses the default Object type. We could add JsonCommandWithPayload a parameter that will cause the RavenCommandResposeType to be using the right response type, but using the StatusCode for this matter looks safer.